server/server.go

```diff
@@ -91,6 +91,7 @@ type Server struct {
 func NewServer(cfg *Config) (*Server, error) {
 	s, err := CreateServer(cfg)
 	if err != nil {
+		s.Close()
```

Comment:
No, I mean if it can returns error, `s` should be nil.
A good coding style should not let the caller cleanup for the callee, if an error occurs inside `CreateServer`, it should cleanup itself.
However, in this case, `CreateServer` never return error, can we just remove the error?